app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -165,6 +165,32 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
         val activeAccount = accountManager.activeAccount
             ?: return // will be redirected to LoginActivity by BaseActivity
 
+        // As early as possible register to be notified when the service saves a "couldn't upload" draft.
+        // This will be called when the app first comes up, which is desirable if a draft has been waiting awhile.
+        accountManager.draftsNeedUserAlert.observe(this) { count ->
+            Log.d(TAG, "draftsNeedUserAlert 2: Notification-worthy draft count " + count)
+            if (count > 0) {
+                AlertDialog.Builder(this)
+                    .setTitle(R.string.action_post_failed)
+                    .setMessage(
+                        if (count == 1)
+                            getString(R.string.action_post_failed_detail)
+                        else
+                            getString(R.string.action_post_failed_detail_plural, count)
+                    )
+                    .setPositiveButton(R.string.action_post_failed_show_drafts) { _: DialogInterface?, _: Int ->
+                        accountManager.clearDraftsAlert() // User looked at drafts
+
+                        val intent = DraftsActivity.newIntent(this)
```

Comment:
Could this be re-worked so that it takes the user to the specific draft that failed, so they only have to press one more button to resend it?

As is, it looks as though they'll need to figure out which draft failed (with nothing in the UI to tell them), open it, and try and re-send it.

E.g., in the DB query instead of returning the count of failed-to-send posts, return a list of the IDs. Then instead of calling `DraftsActivity.newIntent()`, create a `ComposeOptions` with the draft ID, and start `ComposeActivity` instead (see `DraftActivity.onOpenDraft()`, which you might be able to make public)

Reply:
They will be told by the large existing banner reading "this status failed to send".

Although it would be nice if this PR could scroll the drafts view to the bottom so they are more likely to see the exact correct draft.

I don't think putting them in compose view is optimal because then they can't get out without making an affirmative decision to send or not send and also media will begin uploading immediately. Going to the drafts list lets them decide to discard without uploading anything if they choose.

[Screenshot_20230104-071742](https://user-images.githubusercontent.com/277318/210553459-83f4ffbc-325f-4988-a982-2d47167f9fe6.png)